packages/plasma-new-hope/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,222 @@
+import { base as segmentationCSS } from './variations/_segmentation/base';
+import { base as disabledCSS } from './variations/_disabled/base';
+
+export const numberInputRoot = (Root: RootProps<HTMLDivElement, NumberInputRootProps>) =>
+    forwardRef<HTMLInputElement, NumberInputProps>(
+        (
+            {
+                className,
+                style,
+                width,
+                value,
+                min,
+                max,
+                step = 1,
+                isLoading,
+                loader,
+                size,
+                view,
+                clear,
+                shape,
+                segmentation,
+                inputBackgroundType = 'fill',
+                disabled,
+                textBefore,
+                textAfter,
+                customIncrementButton,
+                incrementIcon,
+                customDecrementButton,
+                decrementIcon,
+                isManualInput,
+                onChange,
+                onIncrement,
+                onDecrement,
+                ...rest
+            },
+            ref,
+        ) => {
+            const [innerValue, setInnerValue] = useState<number | string>(value ?? min ?? 0);
+            const [isInputFocused, setIsInputFocused] = useState(false);
+            const [isAnimationRun, setIsAnimationRun] = useState(false);
+
+            const innerWidth = width ? getSizeValueFromProp(width) : '100%';
+
+            const actionIconSize = size === 'xs' ? 'xs' : 's';
+            const solidViewClass = segmentation === 'solid' ? classes.solidView : undefined;
+            const segmentedViewClass = segmentation === 'segmented' ? classes.segmentedView : undefined;
+            const clearViewClass = clear ? classes.clearView : undefined;
+            const disabledClass = disabled ? classes.disabled : undefined;
+            const loadingClass = isLoading ? classes.loading : undefined;
+            const focusedClass = isInputFocused ? classes.focused : undefined;
+
+            const isMinValue = (currentValue: number) => min !== undefined && currentValue <= min;
+            const isMaxValue = (currentValue: number) => max !== undefined && currentValue >= max;
+
+            const decrementButtonDisabled =
+                (isMinValue(Number(innerValue)) && !isAnimationRun) || isLoading
+                    ? classes.actionButtonDecrementDisabled
+                    : undefined;
+            const incrementButtonDisabled =
+                (isMaxValue(Number(innerValue)) && !isAnimationRun) || isLoading
+                    ? classes.actionButtonIncrementDisabled
+                    : undefined;
+
+            const handleDecrement = () => {
+                if (isLoading || disabled || isAnimationRun) {
+                    return;
+                }
+
+                const diffValue = Number(innerValue) - step;
+                const resValue = min && isMinValue(diffValue) ? min : diffValue;
```

Comment:
Тут только все равно надо явно на undefined проверить, потому что при `min=0` будет баг.

```suggestion
                const resValue = min !==undefined && isMinValue(diffValue) ? min : diffValue;
```

И с `max` аналогично

https://github.com/user-attachments/assets/fa1f1ff5-4a93-42c3-9698-87671aecf854

Reply:
Да, я исправил это, теперь ]вно проверку делаю